cmd/geth/snapshot.go

```diff
@@ -78,6 +83,26 @@ WARNING: It's necessary to delete the trie clean cache after the pruning.
 If you specify another directory for the trie clean cache via "--cache.trie.journal"
 during the use of Geth, please also specify it here for correct deletion. Otherwise
 the trie clean cache with default directory will be deleted.
+`,
+			},
+			{
+				Name:     "prune-block",
+				Usage:    "Prune block data offline",
+				Action:   utils.MigrateFlags(pruneBlock),
+				Category: "MISCELLANEOUS COMMANDS",
+				Flags: []cli.Flag{
+					utils.DataDirFlag,
+					utils.AncientFlag,
+					utils.BlockPruneQuantity,
+				},
+				Description: `
+geth offline prune-block for block data in ancientdb.
```

Comment:
is there any side effect that we can not query old blocks any more

Reply:
Have added plus feature for specifying the amount of old blocks that we want to prune, I think this will avoid the side effect you mentioned.